### PR TITLE
Using Symmetric and bunchkaufman in OK

### PR DESCRIPTION
**Pull request by exepulveda**
The pending improvement to OK using Symmetric linear systems

### CODE REVIEW

File: src/estimators/ordinary_kriging.jl [hunk 1]
```diff
@@ -30,7 +30,7 @@ function set_constraints_lhs!(estimator::OrdinaryKriging, LHS::AbstractMatrix, X
   nothing
 end
 
-factorize(estimator::OrdinaryKriging, LHS::AbstractMatrix) = lu(LHS, check=false)
+factorize(estimator::OrdinaryKriging, LHS::AbstractMatrix) = bunchkaufman(Symmetric(LHS), check=false)
```
Comment (juliohm) [inline src/estimators/ordinary_kriging.jl:33]: Nice, did you notice improvements compared to LU factorization? Should we also use this factorization in other variants of Kriging that also have symmetric matrices?

**Comment by juliohm** [comment]
Fixes https://github.com/JuliaEarth/GeoStats.jl/issues/97

File: src/estimators/ordinary_kriging.jl (hunk 1 above)
Comment (exepulveda) [reply src/estimators/ordinary_kriging.jl:33]: Yes, but only with large datasets and > 50 neighbours.
All kriging systems are symmetric. For SO we already have the cholesky factorisation, now for OK the bunchkaufman factorisation. Let me check others. I assume that when building the linear system we are not calculating the "second" have.

Comment (juliohm) [reply src/estimators/ordinary_kriging.jl:33]: Also worth checking the list of available factorizations in the latest versions of the language :+1:

This is what I get in Julia v1.5:

```julia
julia> using LinearAlgebra

julia> subtypes(Factorization)
19-element Array{Any,1}:
 BunchKaufman
 Cholesky
 CholeskyPivoted
 Eigen
 GeneralizedEigen
 GeneralizedSVD
 GeneralizedSchur
 Hessenberg
 LDLt
 LQ
 LU
 LinearAlgebra.QRCompactWY
 QR
 QRPivoted
 SVD
 Schur
 SuiteSparse.CHOLMOD.Factor
 SuiteSparse.SPQR.QRSparse
 SuiteSparse.UMFPACK.UmfpackLU
```

Comment (exepulveda) [reply src/estimators/ordinary_kriging.jl:33]: I already did its for this PR.

**Comment by juliohm** [comment]
Let me know if you would like to add the same changes to the other Kriging variants before we merge the PR.

**Comment by exepulveda** [comment]
Better to close and merge this PR. More changes require ore time to understand and do proper tests.

**Comment by juliohm** [comment]
It would be great if we could have similar modifications tested for the other Kriging variants before merging. I've never used the Bunch-Kaufman factorization myself, and a single commit similar to what you did for OK will work for UK and EDK, makes sense?

**Comment by juliohm** [comment]
If you push this commit, the bot will run the tests on the cloud and we will be able to see if everything passes.

**Comment by exepulveda** [comment]
What a such privilege :-)

**Comment by juliohm** [comment]
Yep. I love this bot :)

**Comment by juliohm** [comment]
Hi Ezequiel why you merged the branch?

**Comment by juliohm** [comment]
I thought we would first make similar changes to the other Kriging variants?

**Comment by exepulveda** [comment]
I think it was a misunderstanding.
I interpreted your comment as "please, merge it". I was wrong. Sorry for the inconvenience. 
I should wait until you merge, always.

Maybe I need to understand better how you want to operate with PR, merges and what to include.

**Comment by juliohm** [comment]
I think we have a method up there in the PR to request review and approve
the merge. When it is ready to merge I'll signal it via the github api.
Thank you for understanding. We need to be careful with what we merge,
specially these estimators that can be used in many places.

**Comment by juliohm** [comment]
Can you please submit another PR for consistency now? One that does similar
changes to other variants.

**Comment by exepulveda** [comment]
Sorry, but my dev environment is at office. I can do it tomorrow morning.

**Comment by juliohm** [comment]
I reverted to LU factorization on the master branch until we have evidence that this Bunch-Kaufman factorization is superior. From the PR it is not clear that the factorization is more stable nor more performant. We need some benchmark results in order to consider it.

**Comment by juliohm** [comment]
If you have some theoretical reason to prefer Bunch-Kaufman, please state it explicitly in the next PR and we can review it in more detail.

**Comment by exepulveda** [comment]
No worries @juliohm. We can continue using lu factorisation. I will be focused on testing the solvers and implementing the basic things that we need to make JuliaEarth a real alternative to use in mining applications. The symmetric factorisation is a minor element although when you work with models of hundreds of millions elements every second counts.

**Comment by juliohm** [comment]
The idea is to improve performance as much as possible. If you have good reasons to choose the other factorization, that is more than welcome. :+1: We just need to be careful before committing such fundamental change. Factorization methods can get quite unstable sometimes if the matrices are ill-conditioned. I will try to find the time to review the Bunch-Kaufman factorization and if you have some material in mind, please don't hesitate to share.

**Comment by juliohm** [comment]
Wait, Bunch-Kaufman is the same as LDL' factorization?

**Pull request merged**